diff --git a/bsp/drivers/usb/host/Makefile b/bsp/drivers/usb/host/Makefile
index a8e5f2c..ce1c2a9 100644
--- a/bsp/drivers/usb/host/Makefile
+++ b/bsp/drivers/usb/host/Makefile
@@ -3,6 +3,7 @@
 # Makefile for the usb host drivers.
 #
 ccflags-y += -I $(srctree)/drivers/usb/host
+ccflags-y += -I $(srctree)/bsp/drivers/usb/host
 
 obj-$(CONFIG_USB_EHCI_HCD_SUNXI)	+= ehci-sunxi.o
 obj-$(CONFIG_USB_OHCI_HCD_SUNXI)	+= ohci-sunxi.o
